glib/src/translate.rs

```diff
@@ -340,12 +340,29 @@ impl IntoGlib for Ordering {
     }
 }
 
+impl<O, E, G> IntoGlib for Result<O, E>
+where
+    G: Copy,
+    O: IntoGlib<GlibType = G>,
+    E: IntoGlib<GlibType = G>,
```

Comment:
My main concern here would be that this requires both variants to be disjoint. If you have a `Result<SomeEnum, SomeOtherEnum>` this would not work correctly, for example :)

Reply:
Initially, I wanted to tie this to both types implementing `From<FlowReturn>` for instance and forcing the result as the `GlibType` for `FlowReturn`. But then the trait system considers it is not constrained enough. Which I vaguely understood as there might be several candidates for this which would not allow to select one and only impl.

So this is the best solution I came up with.

Reply:
To clarify :). I wanted this:

```rust
impl<O, E, F> IntoGlib for Result<O, E>
where
    F: IntoGlib + From<Result<O, E>>,
{
    type GlibType = <F as IntoGlib>::GlibType;

    #[inline]
    fn into_glib(self) -> Self::GlibType {
        F::from(self).into_glib()
    }
}
```

And it leads to that:

```
error[E0207]: the type parameter `F` is not constrained by the impl trait, self type, or predicates
   --> glib/src/translate.rs:362:12
    |
362 | impl<O, E, F> IntoGlib for Result<O, E>
    |            ^ unconstrained type parameter
```

Reply:
Yeah that's exactly why I don't understand why your simplification of `FromValue` worked :)

Reply:
Yes, I guess the difference is that I used the "free" type parameter in an expression such as `IntoGlib<GlibType = G>`.